test/api/controllers/document.test.js

```diff
@@ -540,6 +540,61 @@ describe('DocumentController', () => {
     });
   });
 
+  describe('#updateByQuery', () => {
```

Comment:
missing tests verifying that the controller catch badly formatted requests, that the "refresh" option is correctly forwarded to the ES layer, and that the "source" option correctly removes document contents if set to false (while ensuring that the notifier module still receives them)

Reply:
Since this implementation of `updateByQuery` calls `mUpdate`, the test for the `refresh` option already exists no ?
https://github.com/kuzzleio/kuzzle/blob/master/test/services/elasticsearch.test.js#L2519
+
https://github.com/kuzzleio/kuzzle/pull/1559/files#diff-19c0f27b8ad55ff3c5a5fdf9bd1247dbR919

Reply:
So you should test that the `mUpdate` method is called with the right arguments

Reply:
@scottinet for the test about `source` correctly removes document contents if set to `false`, the test is confusing: 
- I first call `updateByQuery` with `_source` at false, so tthe documents contents are not in the response
- Then It should call `notifyDocumentMChanges` with the document contents
- But it feels like it works with a reference of the result of `updateByQuery`, so for mocha, `notifyDocumentMChanges` is called without the document contents

Reply:
@Aschen it's done here https://github.com/kuzzleio/kuzzle/pull/1559/files#diff-19c0f27b8ad55ff3c5a5fdf9bd1247dbR919

Reply:
You can stub notifyDocumentMChanges and make it clone the received object. You can then use the cloned object to verify whether it correctly receives the complete documents despite having a "_source = false" option passed to the API route.